### PR TITLE
Null value in string array in DataRow causes DataRowComparer to throw NRE, but only if it is the first parameter

### PR DESCRIPTION
**Pull request by AlexG2017**
Use object.Equals() to compare array elements and avoid NRE

### CODE REVIEW

File: src/System.Data.DataSetExtensions/src/System/Data/DataRowComparer.cs [hunk 1]
```diff
@@ -101,7 +101,7 @@ private static bool CompareArray(Array a, Array b)
 
             for (int i = 0; i < a.Length; ++i)
             {
-                if (!a[i].Equals(b[i]))
+                if (!Equals(a[i], b[i]))
```
Comment (benaadams) [inline src/System.Data.DataSetExtensions/src/System/Data/DataRowComparer.cs]: Would could boxing e.g. if `TElem` was `int`

Perhaps better would be?
```csharp
if (!a[i]?.Equals(b[i]) ?? b != null)
```

**Comment by ahsonkhan** [comment]
@dotnet-bot test NETFX x86 Release Build

**Comment by danmosemsft** [comment]
@saurabh500 could you please take a look ?

@dotnet-bot test this please

**Comment by saurabh500** [comment]
@danmosemsft I will be able to take care of this PR tomorrow.

**Comment by saurabh500** [comment]
@AlexG2017 Thanks for the Fix, can you also post the NRE stack ?

**Comment by AlexG2017** [comment]
@saurabh500 please take a look NRE stack trace below
`System.NullReferenceException : Object reference not set to an instance of an object.
at System.Data.DataRowComparer.CompareEquatableArray[TElem](TElem[] a, TElem[] b) in D:\Repositories\dotNet\corefx\src\System.Data.DataSetExtensions\src\System\Data\DataRowComparer.cs:line 104
   at System.Data.DataRowComparer.CompareArray(Array a, Array b) in D:\Repositories\dotNet\corefx\src\System.Data.DataSetExtensions\src\System\Data\DataRowComparer.cs:line 79
   at System.Data.DataRowComparer.AreEqual(Object a, Object b) in D:\Repositories\dotNet\corefx\src\System.Data.DataSetExtensions\src\System\Data\DataRowComparer.cs:line 34
   at System.Data.DataRowComparer`1.Equals(TRow leftRow, TRow rightRow) in D:\Repositories\dotNet\corefx\src\System.Data.DataSetExtensions\src\System\Data\DataRowComparer.cs:line 164`

**Comment by saurabh500** [comment]
@dotnet-bot Test NETFX x86 Release Build please

**Comment by karelz** [comment]
I don't see any failures in [NETFX x86 Release Build results](https://ci3.dot.net/job/dotnet_corefx/job/master/job/windows-TGroup_netfx+CGroup_Release+AGroup_x86+TestOuter_false_prtest/9911/), yet it shows as failing. Restarting the run again ...

@dotnet-bot Test NETFX x86 Release Build please

**Comment by karelz** [comment]
Now I get 'Not Found' - see [details](https://ci3.dot.net/job/dotnet_corefx/job/master/job/windows-TGroup_netfx+CGroup_Release+AGroup_x86+TestOuter_false_prtest/10010/). One last try ...

@dotnet-bot Test NETFX x86 Release Build please

**Comment by saurabh500** [comment]
@dotnet-bot Test NETFX x86 Release Build please

**Comment by saurabh500** [comment]
@dotnet-bot Test this please

File: src/System.Data.DataSetExtensions/tests/System/Data/DataRowComparerTests.cs [hunk 2]
```diff
@@ -298,11 +298,14 @@ public void Equals_NullStringValueInStringArrayInRightColumn_ThrowsNullReference
...
-            Assert.Throws<NullReferenceException>(() => DataRowComparer.Default.Equals(row2, row1));
-            Assert.Throws<NullReferenceException>(() => DataRowComparer.Default.Equals(row1, row3));
+            Assert.True(DataRowComparer.Default.Equals(row1, row2));
+            Assert.True(DataRowComparer.Default.Equals(row2, row1));
 
+            Assert.False(DataRowComparer.Default.Equals(row1, row3));
             Assert.False(DataRowComparer.Default.Equals(row3, row1));
+
+            Assert.False(DataRowComparer.Default.Equals(row2, row3));
+            Assert.False(DataRowComparer.Default.Equals(row3, row2));
```
[the remark this answers is not in the document]
Comment (danmosemsft) [reply src/System.Data.DataSetExtensions/tests/System/Data/DataRowComparerTests.cs]: @AlexG2017 eg `[SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "reason goes here")]`

**Comment by danmosemsft** [comment]
@AlexG2017 chasing up old PR's, is the next action on you here or us?

**Comment by danmosemsft** [comment]
Thanks @AlexG2017 !

**Pull request merged**